n26/api.py

```diff
@@ -295,6 +304,84 @@ def _do_request(self, method: str = GET, url: str = "/", params: dict = None,
+        key64 = base64.b64encode(key)
+        iv64 = base64.b64encode(iv)
+        # encode the key and iv as a json string
+        aes_secret = {
+        'secretKey': key64,
+        'iv': iv64,
+        }
+        # json string has to be represented in byte form for encryption
+        # simplejson used instead of json since regular json can handle only strings, not base64 encoded data
+        unencrypted_aes_secret = bytes(simplejson.dumps(aes_secret), 'utf-8')
+        # Encrypt the secret JSON with RSA using the provided public key
+        public_key = self.get_encryption_key()
+        public_key_non64 = base64.b64decode(public_key['publicKey'])
+        public_key_object = RSA.importKey(public_key_non64)
+        public_key_cipher = PKCS1_v1_5.new(public_key_object)
+        encrypted_secret = public_key_cipher.encrypt(unencrypted_aes_secret)
+        encrypted_secret64 = base64.b64encode(encrypted_secret)
+        # Encrypt user's pin
+        private_key_cipher = AES.new(key=key, mode=AES.MODE_CBC, iv=iv)
+        # the pin has to be padded and transformed into bytes for a correct ecnryption format
+        encrypted_pin = private_key_cipher.encrypt(pad(bytes(pin, 'utf-8'), 16))
+        encrypted_pin64 = base64.b64encode(encrypted_pin)
+
+        return encrypted_secret64, encrypted_pin64
+
+    def create_transaction(self):
+        """
+        Creates a bank transfer order
+
+        All the input should be provided by the user via the command line
+        """
+        # Get all the necessary transfer information from user's input
+        iban = input("Please enter recipient's IBAN (spaces are allowed): ")
+        bic = input("Please enter recipient's BIC: ")
+        name = input("Please enter recipient's bank name: ")
+        reference = input("Please enter transfer reference (optional): ")
+        amount = input("How much would you like to transfer? (only numeric amount, dot separated) ")
+        pin = input("Please enter your pin: ")
+        # Prepare all the data to be transferred to N26
+        encrypted_secret, encrypted_pin = self.encrypt_user_pin(pin)
+        access_token = self.get_token()
+        headers = {'Authorization': 'Bearer {}'.format(access_token),
+                   'encrypted-secret': encrypted_secret,
+                   'encrypted-pin': encrypted_pin}
+        json = {
+            "transaction":{
+                "amount": amount,
+                "partnerBic": bic,
+                "partnerIban": iban,
+                "partnerName": name,
+                "referenceText": reference,
+                "type": "DT"}
+                }
+        url = create_request_url(BASE_URL_DE + '/api/transactions')
+        response = requests.post(url, headers=headers, json=json)
+        response.raise_for_status()
+        # some responses do not return data so we just ignore the body in that case
+        if len(response.content) > 0:
+            return response.json()
```

Comment:
If I understand this correctly, the only difference from `_do_request` here are the additional headers.
Instead of copying the code from `_do_request` I think we can extend the `_do_request` method to accept additional, optional headers that are merged with the `Authorization` header.

Reply:
Yes, that indeed sounds better, I will extend the `_do_request` then.